src/cmd/build.js

```diff
@@ -60,8 +61,21 @@ function defaultPackageCreator(
           filter: (...args) => fileFilter.wantFile(...args),
         })
         .then((buffer) => {
+          let messageData: any;
+          if (manifestData.default_locale) {
+            messageData = readFileSync(
+               path.join(sourceDir, '_locales', manifestData.default_locale,
+                'messages.json'));
+          }
+          let extensionName: string;
+          if (messageData) {
+            extensionName = JSON.parse(messageData).extensionName.description;
```

Comment:
@jtojnar oh I see! I misread your comment at first. 

@shubheksha yes, your patch needs to be updated with a regular expression [like this one](https://github.com/mozilla/web-ext/pull/487#discussion_r79083775) so that it can support more than just the `extensionName` key.